lib/rucio/extensions/forecast.py

```diff
@@ -0,0 +1,113 @@
+# Copyright European Organization for Nuclear Research (CERN)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Authors:
+# - Joaquin Bogado, <jbogadog@cern.ch>, 2017
+
+"""
+Forecast library to predict the Transfers Time To Complete (T3C)
+"""
+
+import json
+import numpy as np
+import requests
+import time
+
+
+class T3CModel():
+    def __init__(self, path=''):
+        try:
+            self.model = json.load(open(path, 'r'))
+        except IOError:
+            self.model = {"STANDARD_LINK": {"r2": 0.0,
+                                            "rmse": 0.0,
+                                            "rate": 23399639.38262837,
+                                            "datalen": 0.0,
+                                            "overhead": 12.025538382153206,
+                                            "diskrw": 12046990.897099394}}
+
+        # Create RSE to site name dictionary
+        r = requests.get('http://atlas-agis-api.cern.ch/\
```

Comment:
Few thing about this:
- The url should probably be put into the config somewhere.
- We should also think about the generic case for this, which should not be much work to implement. What I am saying is, the t3c will be interesting for non-ATLAS Rucio users too, but right now your setup basically requires AGIS.
- I think we should cache this information somewhere too. Depending on where/how this object is constructed, we could be hammering AGIS....

Reply:
@jwackito something to keep in mind: right now for ATLAS we do link calculations site-by-site. to have this completely standalone, we need this functionality RSE-to-RSE, or otherwise we are always dependent on an informaiton catalogue. you should keep this in mind when scaling this out for one of the next iterations. right now we focus on getting the ATLAS case to work, but keep it in mind.

Reply:
@mlassnig, @bari12: Actualy if I do this on RSE basis, I will not need AGIS at all. It's clear to me why we need to do this on site level for ATLAS, but I do not know if there is any real difference doing it one way or the other.
In order to calculate the model for the network time, I think I can even do it on the fly from Rucio DB instead of calculate it from elasticsearch. But to do it at site level need some mapping from AGIS. I will check other daemons to see where they get AGIS data from...

Reply:
site names are in rucio, e.g. rse_attributes. no need to have this url dependency here.

Reply:
@vingar: Then I can calculate everything from Rucio DB. Thanks!